packages/core/src/queue/event-queue.ts

```diff
@@ -265,6 +266,10 @@ export abstract class CoreEventQueue<
       throw new Error('Not ready')
     }
 
+    if (ctx.attempts > 1) {
+      this.emit('delivery_retry', ctx)
```

Comment:
How valuable is this event? Most (all?) of the delivery and segment.io plugins handle their own retries separate from the main event queue, and I think only `before` plugins/source middleware can trigger a pipeline retry.

Reply:
It exists just to make the Inspector trace look linear and chorological

![image](https://user-images.githubusercontent.com/11316020/211664450-48b65bab-e5de-486c-9368-1ed91cff3b36.png)

It will go away when we fix retries once and for all

Reply:
At least for now, regardless of what real value the "retry" mechanism provides, just because it exists, we need to accommodate for it. Part of the reason that explains inclinations towards removing the "global retries"